### PR TITLE
add kms decryption

### PR DESCRIPTION
**Pull request by MancunianSam**
- Decrypt environment variables with kms
- Decrypt the environment variables using kms
- Pass the endpoint to the kms client

### CODE REVIEW

File: src/test/resources/application.conf [hunk 1]
```diff
@@ -1,13 +1,19 @@
 url {
-    api = "http://localhost:9001/graphql"
-    auth = "http://localhost:9002/auth"
+    api = "aHR0cDovL2xvY2FsaG9zdDo5MDAxL2dyYXBocWw=" # http://localhost:9001/graphql
```
Comment (TomJKing) [inline src/test/resources/application.conf:2]: Is it worth adding a bit more information to the comment to explain that the value is the base 64 string?

File: src/main/scala/uk/gov/nationalarchives/api/update/ApiUpdate.scala [hunk 2]
```diff
@@ -11,12 +11,11 @@ import scala.concurrent.{ExecutionContext, Future}
 import scala.language.postfixOps
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/api/update/ApiUpdate.scala:12]: Unnecessary line break

File: src/main/scala/uk/gov/nationalarchives/api/update/ApiUpdate.scala [hunk 3]
```diff
@@ -11,12 +11,11 @@ import scala.concurrent.{ExecutionContext, Future}
...
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/api/update/ApiUpdate.scala:11]: Unused import: `import com.typesafe.config.ConfigFactory`

File: src/main/scala/uk/gov/nationalarchives/api/update/Processor.scala [hunk 4]
```diff
@@ -19,12 +19,12 @@ import scala.concurrent.{Await, ExecutionContext, Future}
 import scala.language.postfixOps
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/api/update/Processor.scala:20]: Unnecessary line break

File: src/main/scala/uk/gov/nationalarchives/api/update/Processor.scala [hunk 5]
```diff
@@ -19,12 +19,12 @@ import scala.concurrent.{Await, ExecutionContext, Future}
...
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/api/update/Processor.scala:19]: Unused imports:

- `import com.amazonaws.services.lambda.runtime.events.SQSEvent`
- `import io.circe`
- `import io.circe.parser.decode`
- `import scala.concurrent.{Await, ...}`

File: src/test/scala/uk/gov/nationalarchives/api/update/ApiUpdateTest.scala [hunk 6]
```diff
@@ -22,8 +22,10 @@ import scala.reflect.ClassTag
 class ApiUpdateTest extends ExternalServicesTest with MockitoSugar with EitherValues {
```
Comment (TomJKing) [inline src/test/scala/uk/gov/nationalarchives/api/update/ApiUpdateTest.scala:22]: Unused import: `import com.typesafe.config.ConfigFactory`

**Review by TomJKing — COMMENTED** [review]
Looks good. Just some minor comments

File: src/test/resources/application.conf (hunk 1 above)
Comment (MancunianSam) [reply src/test/resources/application.conf:2]: I've added in the same comment as for the file-format project.

File: src/main/scala/uk/gov/nationalarchives/api/update/ApiUpdate.scala (hunk 2 above)
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/api/update/ApiUpdate.scala:12]: I always miss the ones that aren't at the end of the file :smile:

File: src/main/scala/uk/gov/nationalarchives/api/update/ApiUpdate.scala (hunk 3 above)
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/api/update/ApiUpdate.scala:11]: I've taken this out.

**Review by TomJKing — APPROVED** [review]

**Pull request merged**